File: NumberGuesser.py
#random number library
import random
print("Number Guesser")
confirm = True
while confirm == True:
    #Range of number
    startNumber = 0
    endNumber = 0
    while True:
        try:
            startNumber = int(input("Input starting range of Number to be generated: "))
            if startNumber <= 0:
                print("Input number greater than 0")
                continue
            endNumber = int(input("Input ending range of Number to be generated: "))
            if endNumber <= 0:
                print("Input number greater than 0")
                continue
            break
        except:
            print("Input number only!")
    #Generate random number
    randomNumber = random.randint(startNumber, endNumber)
    #Total Tries
    totalTries = 0
    while True:
        try:
            totalTries = int(input("Input total tries: "))
            if totalTries <= 0:
                print("Input number greater than 0")
                continue
            break
        except:
            print("Input number only")
    #Let user guess/Game starts
    tries = 1
    number = 0
    while tries <= totalTries:
        try:
            tries +=1
            number = int(input("Input number guess: "))
            if(number == randomNumber):
                print("You guessed the number!")
                break
            elif number < randomNumber:
                print("Random number is greater than that")
            elif number > randomNumber:
                print("Random number is greater than that")
        except:
            tries -=1
            print("Input number only!")
    #Display Failed if user failed to guess the right answer
    if tries == totalTries or number != randomNumber:
        print("You did not get the right number, number was %d"%(randomNumber))
    #Let user play again
    while True:
        tryAgain = input("Want to play again?(Y/N): ").strip().lower()
        if tryAgain == "y":
            continue
        elif tryAgain == "n":
            print("Thanks for playing")
            confirm = False
            break
        else:
            print("Input Y/N only!")

File: NumberGuesser.java
//Import Scanner for user input
import java.util.InputMismatchException;
import java.util.Scanner;
import java.util.Random;
public class NumberGuesser {
    public static void main(String[]args){
        //Scanner and random object
        Scanner scan = new Scanner(System.in);
        Random random = new Random();
        //Variables for min and max number and boolean to keep the CLI running
        int minNumber = 0;
        int maxNumber = 0;
        boolean running = true;
        int randomNumber = 0;
        int tries = 0;
        //While loop to keep the program running
        while(running == true){
            System.out.println("Number guesser game");
            //Let user input minimum number
            while(true){
                    try{
                        //Let user input min and max number
                        System.out.print("Input minimum number: ");
                        minNumber = scan.nextInt();
                        if(minNumber <= 0){
                            System.out.println("Input number greater than 0/zero");
                            continue;
                        }
                        break;
                    }
                    catch(InputMismatchException error){
                        System.out.println("Input integer only!");
                        scan.next();
                    }
                }
            //Let user input maximum number
                while(true){
                    try{
                        //Let user input min and max number
                        System.out.print("Input maximum number: ");
                        maxNumber = scan.nextInt();
                        if(maxNumber <= minNumber){
                            System.out.println("Input number greater than the minimum Number");
                            continue;
                        }
                        break;
                    }
                    catch(InputMismatchException error){
                        System.out.println("Input integer only!");
                        scan.next();
                    }
                }
                //Generate random number
                randomNumber = random.nextInt(maxNumber - minNumber);
                //let user input maximum number of tries
                while(true){
                    try{
                        System.out.printf("Input maximum of tries: ");
                        tries = scan.nextInt();
                        if(tries <= 0){
                            System.out.println("Input number greater than 0!");
                            continue;
                        }
                        break;
                    }
                    catch(InputMismatchException error){
                        System.out.println("Input integer only!");
                        scan.next();
                    }
                }
                //Try counter
                int tryCounter = 1;
                int guessNumber = 0;
                //start of play
                while(tries >= tryCounter){
                    tryCounter+=1;
                    try{
                        System.out.print("Input guess number: ");
                        guessNumber = scan.nextInt();
                        if(guessNumber < randomNumber){
                            System.out.println("Random number is greater than guess number");
                            guessNumber = 0;
                        }
                        else if(guessNumber > randomNumber){
                            System.out.println("Random number is less than guess number");
                            guessNumber = 0;
                        }
                        else if (guessNumber == randomNumber) {
                            guessNumber = 0;
                            System.out.println("You guessed the right number!");
                            break;
                        }
                    }
                    catch(InputMismatchException error){
                        System.out.println("Input integer only!");
                        scan.next();
                    }
                }
            //Reveal random number
            System.out.printf("Random number was: %d\n",randomNumber);
            String playAgain = "";
            while(true){
                System.out.print("Do you want to play again?(Y/N) ");
                playAgain = scan.nextLine().trim();
                if(playAgain.equalsIgnoreCase("Y")){
                    break;
                }
                else if(playAgain.equalsIgnoreCase("N")){
                    running = false;
                    break;
                }
                else{
                    System.out.println("Input Y/N only!");
                }
            }
        }
        scan.close();
        System.out.println("Thanks for playing!");
    }
}